src/resource/resource-opening-hours/opening-period/OpeningPeriod.tsx

```diff
@@ -109,6 +202,47 @@ export default function OpeningPeriod({
           confirmText="Poista"
         />
       </div>
+      {isOpen && (
+        <div className="date-period-details-container">
```

Comment:
It might be easier to read and test this if this would be an independent component. I would also extract the resolving of nonsupported - logic inside that component.

Reply:
Good idea:

- [ ] Extract date period details into its own component